llama_index/storage/chat_store/redis_chat_store.py

```diff
@@ -0,0 +1,278 @@
+import json
+import logging
+import sys
+from typing import Any, List, Optional
+from urllib.parse import urlparse
+
+from llama_index.bridge.pydantic import Field
+from llama_index.llms import ChatMessage
+from llama_index.storage.chat_store.base import BaseChatStore
+
+try:
+    from redis import Redis
+except ImportError:
+    raise ImportError("Could not import redis")
```

Comment:
This is causing the unit tests to fail on the PR.  Should i move these _get_client methods inside the class?  the pydantic type checker gets mad if this import does not exist at the top level because of the Redis type parameter in the method signature of _check_for_cluster

Reply:
Yea move the imports to where it's used.

But you can keep it for type checking using

```
from typing import TYPE_CHECKING, ...

if TYPE_CHECKING:
  import redis
```

This pattern is used in a few places in the library, very useful 👍

Reply:
OK the issue was that the linter required a return type to be imported for line 161 , but i am wondering if just moving these methods to inside the class will simplify everything

Reply:
You can do both I think? But up to you 👍 the thing above let's you import things purely for type checking only

Reply:
Your change looks good to me though

Reply:
I can take a stab at getting CI to pass as well in a bit 👌